packages/notifications/push/amplify_push_notifications/lib/amplify_push_notifications.dart

```diff
@@ -0,0 +1,235 @@
+        .map((event) {
+      final completionHandlerId = (event['payload']
+          as Map<Object?, Object?>?)?['completionHandlerId'] as String;
+      _methodChannel.invokeMethod('completeNotification', completionHandlerId);
+      // TODO convert raw event to RemotePushMessage
+      return PushNotificationMessage();
+    });
+
+    _onNotificationOpened = _notificationOpenedEventChannel
+        .receiveBroadcastStream('NOTIFICATION_OPENED')
+        .cast<Map<Object?, Object?>>()
+        .map((event) {
+      // TODO convert raw event to RemotePushMessage
+      return PushNotificationMessage();
+    });
+
+    // TODO: Enable launch notification API
+
+    _onLaunchNotificationOpened = _launchNotificationOpenedEventChannel
+        .receiveBroadcastStream('LAUNCH_NOTIFICATION_OPENED')
+        .cast<Map<Object?, Object?>>()
+        .map((event) {
+      // TODO convert raw event to RemotePushMessage
+      return PushNotificationMessage();
+    });
+  }
+
+  final AmplifyLogger _logger =
+      AmplifyLogger.category(Category.pushNotifications)
+          .createChild('AmplifyPushNotification');
+  final ServiceProviderClient _serviceProviderClient;
+
+  late final Stream<String> _onTokenReceived;
+  late final Stream<PushNotificationMessage> _onForegroundNotificationReceived;
+  late final Stream<PushNotificationMessage> _onBackgroundNotificationReceived;
+  late final Stream<PushNotificationMessage> _onNotificationOpened;
+  late final Stream<PushNotificationMessage> _onLaunchNotificationOpened;
+
+  bool _isConfigured = false;
+
+  @override
+  Stream<String> get onTokenReceived => _onTokenReceived;
+
+  @override
+  Stream<PushNotificationMessage> get onNotificationReceivedInForeground =>
+      _onForegroundNotificationReceived;
+
+  @override
+  Stream<PushNotificationMessage> get onNotificationOpened =>
+      _onNotificationOpened;
+
+  @override
+  void onNotificationReceivedInBackground(OnRemoteMessageCallback callback) {
+    throw UnimplementedError(
+      'onNotificationReceivedInBackground() has not been implemented.',
+    );
+  }
+
+  @override
+  Future<void> configure({
+    AmplifyConfig? config,
+    required AmplifyAuthProviderRepository authProviderRepo,
+  }) async {
+    // Parse config values from amplifyconfiguration.json
+    final analyticsConfig = config?.analytics?.awsPlugin;
+    if (analyticsConfig == null) {
+      throw const AnalyticsException('No Pinpoint plugin config available');
+    }
+
+    if (_isConfigured) {
+      return;
+    }
+
+    // Block configure if registering device is not complete
+    final deviceToken = await onTokenReceived.first;
+    await _registerDevice(deviceToken);
+
+    // Initialize listeners
+    onTokenReceived.listen(_tokenReceivedListener);
+    onNotificationReceivedInForeground.listen(_foregroundNotificationListener);
+    onNotificationOpened.listen(_notificationOpenedListener);
+
+    // Initialize Endpoint Client
+    await _serviceProviderClient.init(
```

Comment:
This should come before `_registerDevice` since you're calling `_serviceProviderClient.registerDevice` in there